File: root/rest/src/main/java/org/sc/common/rest/TrailImportDto.java
package org.sc.common.rest;

import lombok.Data;
import org.sc.data.model.TrailClassification;
import org.sc.data.model.TrailStatus;

import java.util.Date;
import java.util.List;

@Data
public class TrailImportDto {
    private final String code;
    private final String name;
    private final String description;
    private final int officialEta;
    private final PlaceRefDto startLocation;
    private final PlaceRefDto endLocation;
    private final List<PlaceRefDto> locations;
    private final List<PlaceRefDto> crossways;
    private final TrailClassification classification;
    private final String country;
    private final List<TrailCoordinatesDto> coordinates;
    private final String maintainingSection;
    private final boolean isVariant;
    private final String territorialDivision;
    private final List<LinkedMediaDto> linkedMediaDtos;
    private final Date lastUpdate;
    private final FileDetailsDto fileDetailsDto;
    private final TrailStatus trailStatus;
}

File: root/common/src/main/java/org/sc/common/rest/TrailImportDto.kt
package org.sc.common.rest

import java.util.*

data class TrailImportDto(
     val name: String,
     val description: String,
     val code: String,
     val startPos: PositionDto,
     val finalPos: PositionDto,
     val locations: List<PositionDto>,
     val coordinates: List<TrailCoordinatesDto>,
     val classification: TrailClassification,
     val country: String,
    val lastUpdate: Date,
     val maintainingSection: String
)